src/ucp/core/ucp_ep.h

```diff
@@ -262,7 +265,10 @@ typedef struct ucp_ep {
  */
 typedef struct {
     uintptr_t                     dest_ep_ptr;   /* Remote EP pointer */
-    void                          *user_data;    /* User data associated with ep */
+    union {
+        void                      *user_data;    /* User data associated with ep */
+        ucp_listener_h            listener;      /* Listener that may be associated with ep */
```

Comment:
Why union? I think the fields can co-exist

Reply:
in the client-server flow, when the client has a large worker address, these two fileds should not co-exist on the server side. do you see a case where it will?

Reply:
user can assign user_data after EP creation with `ucp_ep_modify_nb`, shoul listener be valid then?

Reply:
But this pointer will be passed to error handler even if user has not assigned it.

Reply:
the err_cb and the user_data won't be accessed if the err_cb is NULL which is the case in the listener's ep before the accept callback was invoked (giving the user access to the server's ep)
https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_worker.c#L419